### PR TITLE
add rkyv support

### PR DESCRIPTION
**Pull request by dovahcrow**
This PR add optional rkyv support for Decimal.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -35,6 +35,7 @@ rocket = { default-features = false, optional = true, version = "0.5.0-rc.1" }
 serde = { default-features = false, optional = true, version = "1.0" }
 serde_json = { default-features = false, optional = true, version = "1.0" }
 tokio-postgres = { default-features = false, optional = true, version = "0.7" }
+rkyv = {version = "0.7", optional = true}
```
Comment (paupino) [inline Cargo.toml:38]: nit: can we do a couple of things here:
1. this list is alphabetical - can we put it before `rocket`? The parameters are also alphabeticalized.
2. Can we set `default-features = false` and include any required features explicitly?

**Review by paupino — COMMENTED** [review]
Thanks for the contribution - seems simple enough! There are a few additional changes to get this ready for merge:

1. Can you add a test for this? Possibly follow the same style as `borsh`. You'd also need to add a new test into the Makefile and "turn it on" for CI here: https://github.com/paupino/rust-decimal/blob/master/Makefile.toml#L176
2. Add a section to the readme for this feature. It doesn't have to be much! https://github.com/paupino/rust-decimal/blob/master/README.md#borsh
3. Some minor tweaks as described in the `Cargo.toml` file in regards to `default-features` - it helps us keep build times low.

**Comment by dovahcrow** [comment]
@paupino Thank you for the detailed suggestions! I've revised the PR accordingly. This project is so well maintained! I appreciate all the time & effort you put on this project.

**Comment by dovahcrow** [comment]
rkyv also allows one to specify a safe api with a feature flag (https://github.com/rkyv/rkyv#example). Do you want to enable that by default? 

FYI safe API allows one to deserialize types from bytes with some additional checks (e.g. byte alignment), with some performance sacrifice.

File: Cargo.toml [hunk 2]
```diff
@@ -68,7 +70,7 @@ serde-str = ["serde-with-str"]
 serde-with-arbitrary-precision = ["serde", "serde_json/arbitrary_precision", "serde_json/std"]
 serde-with-float = ["serde"]
 serde-with-str = ["serde"]
-std = []
+std = ["rkyv/std"]
```
Comment (paupino) [inline Cargo.toml:73]: This isn't quite right, but I see what you're trying to do. Perhaps for now, we can move the `rkyv/std` feature under `rkyv` also. It just means that the feature requires the `std` feature. I'll make a code suggestion for above.

File: Cargo.toml [hunk 3]
```diff
@@ -59,6 +60,7 @@ default = ["serde", "std"]
 legacy-ops = []
 maths = []
 maths-nopanic = ["maths"]
+rkyv = ["rkyv/size_32"]
```
Comment (paupino) [inline Cargo.toml:63]:
```suggestion
rkyv = ["rkyv/size_32", "rkyv/std", "std"]
```

**Review by paupino — APPROVED** [review]
One minor change required with the feature flags (`std` isn't quite right).

> rkyv also allows one to use a safe api with a feature flag (https://github.com/rkyv/rkyv#example). Do you want to enable that by default?

Your call! I'm happy with how it is right now but if you think it's a useful feature then feel free to add it.

File: Cargo.toml (hunk 3 above)
Comment (dovahcrow) [reply Cargo.toml:63]: rkyv can be used without std. I added rkyv/std in the std feature of rust_decimal

File: Cargo.toml (hunk 2 above)
Comment (paupino) [reply Cargo.toml:73]:
```suggestion
std = []
```

File: Cargo.toml (hunk 3 above)
Comment (dovahcrow) [reply Cargo.toml:63]: Oh I just say your last comment.

File: Cargo.toml (hunk 2 above)
Comment (dovahcrow) [reply Cargo.toml:73]: I think `std = ["rkyv/std"]` means if the std feature of `rust-decimal` is turned on, also turn on the `std` feature of `rkyv`?

Comment (paupino) [reply Cargo.toml:73]: Unfortunately, it includes the dependency by default too. See this section: https://doc.rust-lang.org/cargo/reference/features.html#dependency-features

I see that `"rkyv?/std"` is _actually_ what we want however it requires a minimum rust version of 1.60. Unfortunately, it'll have to wait until the min version requirements catch up: https://github.com/paupino/rust-decimal#minimum-rust-compiler-version

Comment (dovahcrow) [reply Cargo.toml:73]: I see. right "rkyv?/std" is the correct one.

**Comment by dovahcrow** [comment]
@paupino Fixed the std issue and also added safe API support via the `rkyv-safe` feature flag.

**Pull request merged**